canonical/kubeflow-tensorboards-operator
# feat: add terraform modules for tensorboard controller and web-app charms

Closes #140
Closes #141

Create a `terraform/` directory for each of the charms that hosts their individual Terraform modules. It follows the structure proposed in [this spec](https://docs.google.com/document/d/1EG71A2pJ244PQRaGVzGj7Mx2B_bzE4U_OSqx4eeVI1E/edit) and it is based on what was done in canonical/argo-operators#198.

To test the modules:
- Clone the repository and switch this PR's branch.
- For each charm:
  - `cd` into its directory 
  - First run `tox -e tflint` to ensure that linting is correct
  - Create a juju controller and a model name `kubeflow`
  - Run `terraform apply -var "channel=latest/edge" -var "model_name=kubeflow" --auto-approve`

Expected `juju status`:
```
App                     Version  Status   Scale  Charm                   Channel      Rev  Address        Exposed  Message
tensorboard-controller           waiting      1  tensorboard-controller  latest/edge  358  10.152.183.36  no       installing agent
tensorboards-web-app             waiting      1  tensorboards-web-app    latest/edge  346  10.152.183.53  no       installing agent

Unit                       Workload  Agent  Address       Ports  Message
tensorboard-controller/0*  waiting   idle   10.1.199.196         Waiting for gateway info relation
tensorboards-web-app/0*    blocked   idle   10.1.199.252         Please relate to istio-pilot:ingress
```
